data-prepper-plugins/saas-source-plugins/jira-source/src/test/java/org/opensearch/dataprepper/plugins/source/jira/JiraServiceTest.java

```diff
@@ -1,39 +1,260 @@
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Queue;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
 
+import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
 import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.Mockito.doReturn;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.when;
+import static org.opensearch.dataprepper.plugins.source.jira.utils.Constants.ACCESSIBLE_RESOURCES;
+import static org.opensearch.dataprepper.plugins.source.jira.utils.Constants.BASIC;
+import static org.opensearch.dataprepper.plugins.source.jira.utils.Constants.CREATED;
+import static org.opensearch.dataprepper.plugins.source.jira.utils.Constants.KEY;
+import static org.opensearch.dataprepper.plugins.source.jira.utils.Constants.UPDATED;
 
 
 /**
  * The type Jira service.
  */
-
 @ExtendWith(MockitoExtension.class)
 public class JiraServiceTest {
 
-    Logger log = LoggerFactory.getLogger(JiraServiceTest.class);
+    @Mock
+    private RestTemplate restTemplate;
+
+    @Mock
+    private JiraAuthConfig authConfig;
 
     @Mock
-    RestTemplate restTemplate;
+    private SearchResults mockSearchResults;
+
+    @Mock
+    private StringBuilder jql;
+
+    private static final Logger log = org.slf4j.LoggerFactory.getLogger(JiraServiceTest.class);
+    private JiraSourceConfig jiraSourceConfig;
+    private Queue<ItemInfo> itemInfoQueue;
+    private List<Future<Boolean>> futureList;
+    private ExecutorService crawlerTaskExecutor;
+    private PluginExecutorServiceProvider executorServiceProvider = new PluginExecutorServiceProvider();
+    private JiraService jiraService;
+    private List<String> issueType;
+
+    @BeforeEach
+    void setUp() {
+        itemInfoQueue = new ConcurrentLinkedQueue<>();
+        crawlerTaskExecutor = executorServiceProvider.get();
+        issueType = new ArrayList<>();
+        futureList = new ArrayList<>();
+    }
+
+    @AfterEach
+    void tearDown() {
+        executorServiceProvider.terminateExecutor();
+    }
+
+    @Test
+    void testJiraServiceInitialization() throws JsonProcessingException {
+        jiraSourceConfig = createJiraConfiguration(BASIC, issueType);
+        jiraService = new JiraService(restTemplate, jiraSourceConfig, authConfig);
+        assertNotNull(jiraService);
+    }
+
+    @Test
+    public void testGetJiraEntities() throws JsonProcessingException, InterruptedException, TimeoutException {
+        issueType.add("Task");
```

Comment:
Each test having their own list of issueType is better. One test modifying the list will pollute other test

Reply:
even if I make a new one every setUp?

Reply:
will change anyways

Reply:
ok